libs/execution_context/ebpf_link.c

```diff
@@ -90,10 +90,10 @@ ebpf_link_initialize(
         NULL,
         NULL);
 
-    if (!provider_data) {
```

Comment:
do you need this check in addition to the return value check?

Reply:
Nope.  The `_Outptr_` SAL annotation on the provider_data argument to the above function declares that it cannot be null on success so the new check is sufficient.